lib/ansible/executor/discovery/python.py

```diff
@@ -0,0 +1,57 @@
+# Copyright: (c) 2018 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# FUTURE: this could be swapped out for our bundled version of distro so long as we maintain Py2.6 compat and
+# don't need to add any other code that would necessitate any kind of script assembly
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import json
+import platform
+import re
+
+
+def get_platform_info():
+    distro = None
+    version = None
+
+    if hasattr(platform, 'dist'):
+        distinfo = platform.dist()
+        distro = distinfo[0]
+        version = distinfo[1]
+
+    if not distro or not version:
+        # try to fall back to parsing /etc/os-release if present
+        osr = parse_os_release()
+        distro = osr.get('ID')
+        version = osr.get('VERSION_ID')
```

Comment:
This won't work on Python3 because ```parse_os_release()``` is using byte strings for the key and value but this code will be looking for text versions of the keys.